app/src/main/java/org/simple/clinic/user/UserSession.kt

```diff
@@ -286,6 +288,18 @@ class UserSession @Inject constructor(
     return jsonAdapter.fromJson(error.response().errorBody()!!.source())!!
   }
 
+  fun clearLoggedInUser(): Completable {
+    return loggedInUser()
+        .firstOrError()
+        .filter { it is Just<User> }
+        .map { (user) -> user!! }
+        .flatMapCompletable {
+          Completable.fromAction {
+            appDatabase.userDao().deleteUserAndFacilityMappings(it, appDatabase.userFacilityMappingDao())
```

Comment:
Hrm, I'm not sure what I think about DAOs accessing each other. What was the reason behind doing this?

Reply:
Delete facility mappings should belong to the facility mapping dao, and I needed to run them in a transaction. The other option is to execute them in a database transaction externally, which I don't want to place in the UserSession.